src/ol/interaction/selectinteraction.js

```diff
@@ -68,6 +77,12 @@ ol.interaction.Select = function(opt_options) {
    */
   this.multi_ = goog.isDef(options.multi) ? options.multi : false;
 
+  /**
+   * @private
+   * @type {ol.interaction.SelectFilterFunction}
+   */
+  this.filter_ = goog.isDef(options.filter) ? options.filter : null;
```

Comment:
You could change this to:

```js
  this.filter_ = goog.isDef(options.filter) ? options.filter : goog.functions.TRUE;
```

and thereby avoid testing that `this.filter_` is defined in the `handleEvent` function.